### PR TITLE
SOAP: Support ValueGenerator

### PR DESCRIPTION
**Pull request by ricekot**
- Resolves zaproxy/zaproxy#3345.
soapValueGenerator

Signed-off-by: ricekot <ricekot@gmail.com>

### CODE REVIEW

**Review by kingthorin — APPROVED** [review]
Nice job!

**Review by psiinon — APPROVED** [review]
Very nice!

**Comment by psiinon** [comment]
Oh, would be good to update the help to mention this :)

**Review by kingthorin — APPROVED** [review]
Looks even better! ⚡

**Comment by psiinon** [comment]
Agreed! 👍

File: addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/WSDLCustomParser.java [hunk 1]
```diff
@@ -429,6 +442,37 @@ private void recordOperation(int wsdlID, BindingOperation bindOp) {
...
+                    Control.getSingleton()
+                            .getExtensionLoader()
+                            .getExtension(ExtensionSpider.class)
+                            .getValueGenerator();
+        } catch (NullPointerException e) {
+            LOG.debug(e.getMessage());
+        }
+
+        if (coreValGen != null) {
+            HashMap<String, String> fieldAttributes = new HashMap<String, String>();
```
Comment (thc202) [inline addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/WSDLCustomParser.java:458]: - `Map`
 - The type arguments could be removed.

File: addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/WSDLCustomParser.java [hunk 2]
```diff
@@ -429,6 +442,37 @@ private void recordOperation(int wsdlID, BindingOperation bindOp) {
...
+            fieldAttributes.put("Control Type", "TEXT");
+            fieldAttributes.put("type", name);
+            String valGenValue =
+                    coreValGen.getValue(
+                            null,
+                            null,
+                            name,
+                            "",
+                            Collections.<String>emptyList(),
+                            Collections.<String, String>emptyMap(),
```
Comment (thc202) [inline addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/WSDLCustomParser.java:468]: The type arguments could be removed.

**Comment by thc202** [comment]
Could you add some tests?

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**